tox.ini

```
@@ -1,16 +1,17 @@
 [tox]
-envlist = {py37,py38,py39}-django{22,32},py310-django32
+envlist = {py38,py39,py310}-django{3.2,4.0,4.1,4.2},py310-django42
 
 [gh-actions]
 python =
-    3.7: py37
     3.8: py38
     3.9: py39
     3.10: py310
 
 [testenv]
 deps =
-    factory_boy==2.3.1
-    django22: Django>=2.2,<3.0
+    factory_boy==3.2.1
     django32: Django>=3.2,<4.0
+    django40: Django>=4.0
```

Update these like so, otherwise they will install the latest version of Django:
```suggestion
    django40: Django>=4.0,<4.1
```

---

Updated